standard.md

```diff
@@ -0,0 +1,114 @@
+
+There is no limit to the number of file that can be provided on a single file, although it is suggested
+
+### Required options
+
+- **alias**: `str`. Unique name for the dataset. Any (`alias`, `variable`) pair must be unique
+- **filename**: `str`. Absoulute path to the data file.
+- **variable**: `str`. Variable name to be used by the diagnostic. It does not have to match any of the CF conventions names in the file
+
+### Reserved options
+
+#### Related to the dataset
+
+- **project**: `str`. Name of the project at which this dataset belongs. It can be a real project, like `CMIP5`, `CMIP6` or a tool defined agrupation of datasets like `Observations` or `Reanalysis`
+- **activity**: `str`. CMIP activity or analog for the current data
+- **institute**: `list(str)`. List of institutes associated to the current data.
+- **dataset**: `str`. Name of the dataset, being it the CMIP model indetifier, the name of an observation
+- **ensemble**: `str`. Ensemble identifier for the current data
+
+#### Related to the variable
+
+- **table**: `str`. Original mip of the current variable
+- **frequency**: `str`. Original frequency of the data used
+- **modeling_realm**: `list(str)`. List of realms at which the variable belonh
+- **grid**: `str`. Original grid of the variable
+- **units**: `str`. Units of the data
+- **short_name**: `str`. Variable name used in the file.
+- **standard_name**: `str`. Standard name of the variable.
+- **long_name**: `str`. Long name of the variable
+
+#### Others
+
+- **start**: `str`. Starting date of the data, in the format YYYYMMDD
+- **end**: `str`, Ending date of the data, in the format YYYYMMDD
+- **start_year**: `int`. Starting year of the data
+- **end_year**: `int`. End year of the data
+- **reference_dataset**: `str`. Dataset to be used as the reference to compare the current dataset with, given as the alias.
+
+## The script formal description file
+
+It is recommended that each script comes with a description file in yaml syntax which includes :
+- **script_name**: `str`. A label for the name of the script, which uniquely identifies it, and is delivered by an authoritative entity (such as an ESGF, ENES or WCRP entitiy)
+- **script_interface_version**: `str`. A label for distinguishing among various version of the script interface
+-  **mandatory_keys**: `list(str)`. the list of reserved and custom keys for which the tool must provide values
+- **input_type**: `str`. which can take values 'member', 'ensemble' and 'any', and which allows the tool to check that values provided to the script are of the right type. 'ensemble' means : an ensemble with more than one member, while 'any' allows for an ensemble of size 1. If the type differs among the variables, this entry can be a dictionary with key = variable name
```

Comment:
Could you explain why you think this is needed? From the current text, I do not understand what values are checked and how this helps with checking those.

Reply:
Distinguishing between 'member' and 'ensemble' allows the tool to nicely handle ensembles : when a script of type 'member' is applied to an ensemble, the tool automatically loops on applying the script to each member, and builds a resulting ensemble object with all member results. This of course make sense only for tools which include ensemble objects, and which actually handle script outputs.
Category 'any' is not instrumental, let us forget about it.

Reply:
I agree with @bouweandela, I found the name `input_type` to be confusing.

Reply:
As far as I understand, @bouweandela was not objecting on the key name, but on the key use. I am flexible about the key name

Reply:
Well I found the meaning a bit confusing as well. But after reading your explanation and how `input_type` is only needed for tools that include ensemble objects, maybe `input_type` (and `can_select` for that matter) could be given as examples of `mandatory_keys`. Rather than having their own entry, since it's information that only needs to be present for certain tools.

Reply:
It really cannot be an example of a mandatory key, as it is not a value that would change among script calls; it is rather a property of the script.
I propose to rename the key and change the description to make things more clear : 

> **process_ensembles**: `bool` (optional, defaults to True). Tells the tool if the script can process ensembles. If false, and when the script is to be applied to an ensemble, some tools may take in charge a loop on ensemble members. Please refer to tool documentation.

If the standard interface is to be used by more than one tool, allowing for defining this keyword is instrumental. 

Because I suggest that the key is optional and that the default value matches ESMValTool's behaviour, this has no detrimental impact on ESMValTool.

Reply:
It really cannot be an example of a mandatory key, as it is not a value that would change among script calls; it is rather a property of the script.
I propose to rename the key and change the description to make things more clear : 

> **process_ensembles**: `bool` (optional, defaults to True). Tells the tool if the script can process ensembles. If false, and when the script is to be applied to an ensemble, some tools may take in charge a loop on ensemble members. Please refer to tool documentation.

If the standard interface is to be used by more than one tool, allowing for defining this keyword is instrumental. 

Because I suggest that the key is optional and that the default value matches ESMValTool's behaviour, this has no detrimental impact on ESMValTool.

Reply:
To me they don't seem as standard settings, but rather something very specific to certain tools so I am not sure they should be considered as an standard. 

I am adding them as optional settings as suggested, but maybe it's something that at some point should be revised.

With that, if the wording works for you, I would try to have to have this first draft merged so that the issues can be taken care of.

Reply:
> To me they don't seem as standard setting**s** .. I am adding **them**..

Are we referring to more than one setting ? For me, we refer only to `input_type`. 

I prefer that we change the wording to the last one that I proposed , namely :

> **process_ensembles**: `bool` (optional, defaults to True). Tells the tool if the script can process ensembles. If false, and when the script is to be applied to an ensemble, some tools may take in charge a loop on ensemble members. Please refer to tool documentation.

Reply:
>Are we referring to more than one setting ? For me, we refer only to input_type.

I think we have a similar situation with `can_select`. See https://github.com/IsEnesEval/diagstandards/pull/1#discussion_r747429301

Reply:
For me, it would be much detrimental to the standard if it didn't allow the resources efficiency that `can_select` brings

Reply:
It is included in the standard. Just as an optional parameter.